src/java/fr/paris/lutece/util/signrequest/IPAuthentificator.java

```diff
@@ -0,0 +1,72 @@
+package fr.paris.lutece.util.signrequest;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class IPAuthentificator implements RequestAuthenticator {
+	
+	private List<String> _listIPs;
+	private boolean isWhiteList;
+	
+	/**
+	 * Gets the list of IP adresses
+	 * 
+	 * @return the list of IP adresses
+	 */
+	public List<String> getIPs( ) {
+		return _listIPs;
+	}
+	
+	/**
+	 * Sets the list of IP adresses
+	 * 
+	 * @param list
+	 * 		The list of IP adresses
+	 */
+	public void setIPs( List<String> list ) {
+		this._listIPs = list;
+	}
+	
+	/**
+	 * Gets the boolean that defines if it is a white or black list
+	 * 
+	 * @return the boolean
+	 */
+	public boolean isWhiteList( ) {
+		return isWhiteList;
+	}
+
+	/**
+	 * Sets the boolean that defines if it is a white or black list
+	 * 
+	 * @param isWhiteList
+	 * 		The boolean
+	 */
+	public void setWhiteList( boolean isWhiteList ) {
+		this.isWhiteList = isWhiteList;
+	}
+	
+	/**
+	 * {@inheritDoc }
+	 */
+	@Override
+	public boolean isRequestAuthenticated( HttpServletRequest request ) {
+		if( _listIPs != null && _listIPs.contains( request.getRemoteAddr( ) ) )
```

Comment:
what if the app is behind a proxy which does not put the client IP in the remote Address, but in the X-Forwarded-For header ?

Reply:
This does not match our needs, we are going to use it on known IPs which we are sure won't use a proxy.
Beside the X-Forwarded-For can be spoofed.